src/formatDate.js

```diff
@@ -8,7 +8,44 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const slicedDate = date.split(fromFormat[3]);
+  const newSeparator = toFormat[3];
+
+  const dayIndex = fromFormat.indexOf('DD');
+  const monthIndex = fromFormat.indexOf('MM');
+  const yearIndex = Math.abs(dayIndex + monthIndex - 3);
```

Comment:
Using Math.abs(dayIndex + monthIndex - 3) to find the yearIndex is not intuitive. A more straightforward approach would be to directly search for the 'YYYY' or 'YY' in the fromFormat array.